src/core/opamStd.ml

```diff
@@ -903,19 +903,23 @@ module OpamSys = struct
       let split = if clean then OpamString.split else OpamString.split_delim in
       split path path_sep
 
-  let with_process_in cmd args f =
+  let process_in cmd args =
     if Sys.win32 then
       assert false;
-    let path = split_path_variable (Env.get "PATH") in
-    let cmd =
-      List.find Sys.file_exists (List.map (fun d -> Filename.concat d cmd) path)
-    in
-    let ic = Unix.open_process_in (cmd^" "^args) in
+    let args = Array.of_list (cmd :: args) in
+    let env = Env.raw_env () in
     try
-      let r = f ic in
-      ignore (Unix.close_process_in ic) ; r
-    with exn ->
-      ignore (Unix.close_process_in ic) ; raise exn
+      let path = split_path_variable (Env.get "PATH") in
+      let cmd =
+        List.find OpamStubs.is_executable
+          (List.map (fun d -> Filename.concat d cmd) path)
+      in
+      let (ic, _, _) as p = Unix.open_process_args_full cmd args env in
```

Comment:
```suggestion
      let args = Array.of_list (cmd :: args) in
      let (ic, _, _) as p = Unix.open_process_args_full cmd args env in
```
`cmd` changed but not `args`.

Reply:
i don't think it really matters what value `argv[0]` has for those tools, but i can change that for the sake of having one less root value tracked by the GC for a short period of time